### PR TITLE
fix vasp potential

### LINKED ISSUE #553: VASP Potentials with custom elements - Future warnings
Example code:
```
from pyiron_atomistics import Project
pr = Project("debug")
structure = pr.create.structure.ase.bulk("Al", cubic=True)
element = pr.create_element(
    new_element_name='Al_GW',
    parent_element="Al",
    potential_file='Al_GW'
)
structure[:] = element 

job = pr.create.job.Vasp("test")
job.structure = structure
job.run()
```

I get the following warnings: 
```
/u/system/SLES12/soft/pyiron/dev/anaconda3/lib/python3.8/site-packages/pyiron_atomistics/vasp/potential.py:215: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self._potential_df = self._potential_df.append(ds)
/u/system/SLES12/soft/pyiron/dev/anaconda3/lib/python3.8/site-packages/pyiron_atomistics/vasp/potential.py:217: FutureWarning: The default dtype for empty Series will be 'object' instead of 'float64' in a future version. Specify a dtype explicitly to silence this warning.
  ds = pandas.Series()
/u/system/SLES12/soft/pyiron/dev/anaconda3/lib/python3.8/site-packages/pyiron_atomistics/vasp/potential.py:220: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  self._default_df = self._default_df.append(ds)
```

@sudarsan-surendralal  Can you take a look at this?

### PR DESCRIPTION
**Pull request by jan-janssen**

### CODE REVIEW

**Comment by jan-janssen** [comment]
Somehow `concat` is not the same as `append`:
```
df = pd.DataFrame({"A": [1], "B": [2]})
df.append(pd.Series([3, 4], index=["A", "C"]), ignore_index=True) != pd.concat([df, pd.Series([3, 4], index=["A", "C"])])
```

File: pyiron_atomistics/atomistics/structure/periodic_table.py [hunk 1]
```diff
@@ -350,7 +350,9 @@ def add_element(
         parent_element_data_series["Parent"] = parent_element
         parent_element_data_series.name = new_element
         if new_element not in self.dataframe.T.columns:
-            self.dataframe = self.dataframe.append(parent_element_data_series)
+            self.dataframe = pandas.concat(
+                [self.dataframe, parent_element_data_series],
+            )
```
Comment (sudarsan-surendralal) [inline pyiron_atomistics/atomistics/structure/periodic_table.py]:
```suggestion
            self.dataframe = pandas.concat(
                [self.dataframe, parent_element_data_series.to_frame().T],
            )
```

File: pyiron_atomistics/vasp/potential.py [hunk 2]
```diff
@@ -214,10 +214,13 @@ def add_new_element(self, parent_element, new_element):
...
-            ds.name = new_element
-            ds["Name"] = "-".join(name_list)
-            self._default_df = self._default_df.append(ds)
+            ds = pandas.Series(
+                ["-".join(name_list)],
+                index=["Name", "Filename"],
+                dtype=str,
+                name=new_element,
+            )
+            self._default_df = pandas.concat([self._default_df, ds])
```
Comment (sudarsan-surendralal) [inline pyiron_atomistics/vasp/potential.py]:
```suggestion
            self._default_df = pandas.concat([self._default_df, ds.to_frame().T])
```

**Review by sudarsan-surendralal — CHANGES_REQUESTED** [review]
For `concat` to work, you should convert the series to a dataframe before you apply the function

**Review by sudarsan-surendralal — APPROVED** [review]
After some changes and testing, the changes look okay

**Comment by jan-janssen** [comment]
> After some changes and testing, the changes look okay

Perfect - thanks a lot

**Pull request merged**